### ISSUE TITLE
Outlook on the web: Deleted attachment is restored when sending email

### ISSUE THREAD

**Issue by clavx** [OfficeDev/office-js#1739]
> Message from office-js bot:
We’re closing this issue because it has been inactive for a long time. We’re doing this to keep the issues list manageable and useful for everyone. If this issue is still relevant for you, please create a new issue.
> Thank you for your understanding and continued feedback.

When sending an email using OWA, the attachment is restored and sent.

## Expected Behavior

The attachment list can be obtained correctly.

## Current Behavior

The file is supposed to be deleted, but it is left without being visible.

## Steps to Reproduce, or Live Example

1. Compose an email
2. Select the preset "Fast 3G" from the Network Throttling settings
3. Add three attachments
4. Just before the transfer of the second attachment finishes, delete the first and second attachments.
    - At this time, the status will be 1: Deleting, 2: Canceling.
    - The second attachment will not be displayed on the OWA screen.
5. If you send an email in this state, the second attachment will be restored and attached to the email.

In order to reproduce it, the timing of the cancellation in step 4 is very important.
You need to do it at the timing shown in the attached video.

https://user-images.githubusercontent.com/58804032/111951618-d327bd80-8adb-11eb-97d5-ab74e4b1b21a.mp4

## Context

I investigated the above behavior after receiving a report that attachment deletion by the on-send add-in may not have deleted all attachments.
When I look at the corresponding email in Outlook for Windows, it seems that the attachments that should have deleted are still there, but I can no longer see that from OWA.
It seems to be fixed by closing the open mail completely and reopening it.

## Your Environment

* Platform [PC desktop, Mac, iOS, Office on the web]: PC desktop
* Host [Excel, Word, PowerPoint, etc.]: Outlook
* Office version number: 20210315003.14
* Operating System: Windows 10 20H2
* Browser (if using Office on the web): Edge Version 89.0.774.57

**Comment by ElizabethSamuel-MSFT** [comment]
@clavx Thanks for raising this issue.

@exextoc Can you take a look?

Thanks.

**Comment by clavx** [comment]
The issue could no longer be reproduced by the procedure I first reported.

https://user-images.githubusercontent.com/58804032/114521427-ed177480-9c31-11eb-85ac-79a5a44f6abb.mp4

I was able to reproduce it using a different reproduction procedure, so I will report it here.

1. Compose an email
2. Add 4 attachments
3. Delete the second and third attachments

If you reduce the file size, you can reproduce the issue even if you have low bandwidth.

https://user-images.githubusercontent.com/58804032/114521473-f9033680-9c31-11eb-80cb-6fe2659ea149.mp4

Test ENV: Azure
OWA hostVersion: 20210405005.04

**Comment by exextoc** [comment]
@clavx : Thanks for updated repro steps. The issue has been put on our backlog. We unfortunately have no timelines to share at this point.

**Comment by zorauf** [comment]
@clavx Thanks for reporting. Can you please take a fiddler trace while this issue repros and attach it.
(Please make sure to remove auth headers and cookies from the fiddler trace, if you are sharing using a public mechanism)

**Comment by clavx** [comment]
@zorauf  I captured a fiddler trace.
I disallowed the sign-in of the user who got the trace, so I think it is safe even if the cookie is raw data.

https://cube5w-my.sharepoint.com/:f:/g/personal/test_cube5w_onmicrosoft_com/ErO2rE9sqhpDnJuNCm5tlWABujvotsh5JPV_wOgsNPJfow?e=KKOjlQ

This is an update to the reproduction procedure.
It was not necessary to delete the two files.
It can be reproduced by deleting one of the files when the progress bar shows that two files are being uploaded at the same time.

https://user-images.githubusercontent.com/58804032/116032709-affaab80-a64f-11eb-8908-2ee9e95efc90.mp4

If the line bandwidth is lowered, it will be uploaded one by one as I reported, and I have confirmed that the issue has been resolved.
If the line bandwidth is good, it seems to upload multiple files at the same time, and this issue remains.
